modules/siddhi-core/src/main/java/org/wso2/siddhi/core/aggregation/IncrementalDataPurging.java

```diff
@@ -0,0 +1,366 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.siddhi.core.aggregation;
+
+import org.apache.log4j.Logger;
+import org.wso2.siddhi.core.config.SiddhiAppContext;
+import org.wso2.siddhi.core.event.ComplexEventChunk;
+import org.wso2.siddhi.core.event.state.MetaStateEvent;
+import org.wso2.siddhi.core.event.state.StateEvent;
+import org.wso2.siddhi.core.event.stream.MetaStreamEvent;
+import org.wso2.siddhi.core.event.stream.StreamEventPool;
+import org.wso2.siddhi.core.exception.DataPurgingException;
+import org.wso2.siddhi.core.exception.NoSuchDurationException;
+import org.wso2.siddhi.core.exception.SiddhiAppCreationException;
+import org.wso2.siddhi.core.executor.VariableExpressionExecutor;
+import org.wso2.siddhi.core.table.Table;
+import org.wso2.siddhi.core.util.SiddhiConstants;
+import org.wso2.siddhi.core.util.collection.operator.CompiledCondition;
+import org.wso2.siddhi.core.util.collection.operator.MatchingMetaInfoHolder;
+import org.wso2.siddhi.query.api.aggregation.TimePeriod;
+import org.wso2.siddhi.query.api.annotation.Annotation;
+import org.wso2.siddhi.query.api.annotation.Element;
+import org.wso2.siddhi.query.api.definition.AggregationDefinition;
+import org.wso2.siddhi.query.api.definition.Attribute;
+import org.wso2.siddhi.query.api.definition.TableDefinition;
+import org.wso2.siddhi.query.api.expression.Expression;
+import org.wso2.siddhi.query.api.expression.Variable;
+import org.wso2.siddhi.query.api.expression.condition.Compare;
+
+import java.util.ArrayList;
+import java.util.EnumMap;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * This class implements the logic which is needed to purge data which are related to incremental
+ **/
+public class IncrementalDataPurging implements Runnable {
+    private static final Logger LOG = Logger.getLogger(IncrementalDataPurging.class);
+    private long purgeExecutionInterval = Expression.Time.minute(5).value();
+    private long sec = Expression.Time.sec(30).value();
+    private long mins = Expression.Time.hour(24).value();
+    private long hours = Expression.Time.day(30).value();
+    private long days = Expression.Time.year(5).value();
+    private long months = -1;
+    private long years = -1;
+    private boolean purgingEnabled = true;
+    private Map<TimePeriod.Duration, Long> retentionPeriods = new EnumMap<>(TimePeriod.Duration.class);
+    private StreamEventPool streamEventPool;
+    private Map<TimePeriod.Duration, Table> aggregationTables;
+    private SiddhiAppContext siddhiAppContext;
+    private Map<String, Table> tableMap;
+    private static ScheduledFuture scheduledFuture;
```

Comment:
Let's use a meaningful name here to hint what this 'Future' does

Reply:
fixed